### DISCUSSION TITLE
Navigation shows the wrong number of pictures in Review

### DISCUSSION THREAD

**Discussion by pittbull** [photoprism/photoprism#3866] [Q&A]
For some time now the number of photos to review has been off. As per screenshot it states 20, but when opening the folder there are only 6 photos to review. 

Any way to resolve this? I am using MariaDB as backend. 

![image](https://github.com/photoprism/photoprism/assets/1532149/b99c9316-2a95-4445-8056-7afc7af93e13)

**Comment by lastzero** [comment]
Did you try to reload the UI and if that didn't help to reindex (without full rescan)? Do you see any errors or warnings in the logs or could the files be hidden, e.g. because they could not be converted to JPEG if they have another format? What version are you using? Do you use our Docker image or a third-party integration?

**Reply by lastzero** [reply]
See also this (more general) troubleshooting checklist for possible reasons why files are not displayed as expected:

- https://docs.photoprism.app/getting-started/troubleshooting/#missing-pictures

**Reply by pittbull** [reply]
- Confirm having reset local cache and reloaded UI
- I have done a reindex
- The command "ls -aR | grep "^\."" reveals no hidden files
- Running Build [231103-539e18d98]
- No errors in the log

**Reply by lastzero** [reply]
Can you reproduce this on our public demo at https://demo.photoprism.app/ and/or provide us with affected files so we can take a look at the issue? All of our tests are green and until we know how to reproduce this, we can't troubleshoot and potentially fix it. The information we need can be found here:

- https://www.photoprism.app/kb/reporting-bugs

**Reply by lastzero** [reply]
Also try a full rescan in case your index has inconsistencies. Should this still not work then it would help to know if an older release shows the correct number in the UI as we've recently made readability changes to the queries (so this shouldn't have an impact, but you never know).

**Reply by pittbull** [reply]
I wouldn't be so bold as to claim this is a bug in the software, rather something in the database would be more likely.

I will initiate a full rescan and report the results.

**Reply by pittbull** [reply]
A bit unsure as to how I can reproduce this via the link provided?

**Reply by lastzero** [reply]
We neither, that's the problem :) if it might be something in the database then try to update it and run a full rescan as that may correct it.

**Comment by pittbull** [comment]
Full index running, and so far this is the only error produced:
`time="2023-11-05T17:38:28+01:00" level=debug msg="metadata: runtime error: index out of range [0] with length 0 in 20110719-093215.jpg (exif gps-info)"`

**Comment by pittbull** [comment]
Full scan completed. No other errors than the one above and two mentions of hidden files being included. GUI still showing 14 files, yet when clicked no photos appear.

**Reply by lastzero** [reply]
Did you run it with the cleanup option?

If you know how to [open a terminal](https://docs.photoprism.app/getting-started/docker-compose/#opening-a-terminal) you can additionally try the following commands:

**photoprism purge --hard**

```
NAME:
   photoprism purge - Updates missing files, photo counts, and album covers

USAGE:
   photoprism purge [command options] [arguments...]

OPTIONS:
   --hard  permanently remove from index
   --dry   dry run, don't actually remove anything
```

**photoprism cleanup**

```
NAME:
   photoprism cleanup - Removes orphaned index entries, sidecar and thumbnail files

USAGE:
   photoprism cleanup [command options] [arguments...]

OPTIONS:
   --dry  dry run, don't actually remove anything
```

Furthermore, if you know how to connect to your MariaDB database (for example by using the `mariadb` command in the photoprism container), then you can query the affected rows by searching for images with a quality < 3 (also try excluding -1 as that would indicate a hidden/broken image, not one that can be reviewed), e.g. 

```sql
SELECT * FROM photos WHERE photo_quality > -1 AND photo_quality < 3 AND photo_type = 'image';
```

To open the mariadb console, you can also do the following if you run your instance with Docker Compose:

```
docker compose exec mariadb mariadb -uroot -p photoprism
```

**Reply by pittbull** [reply]
I did not run with cleanup, but can retry.

If I run your select statement it responds with 733 lines.

**Reply by pittbull** [reply]
If I run photoprism cleanup -dry, it lists the following files (14!) would be removed from the index. I've checked and the files are present in the originals folder.

```
INFO[2023-11-06T13:31:30+01:00] cleanup: 2009-08-14/20090814-193950 would be removed from index
INFO[2023-11-06T13:31:30+01:00] cleanup: 2010-08-15/20100815-141700 would be removed from index
INFO[2023-11-06T13:31:30+01:00] cleanup: 2010-10-09/20101009-134039 would be removed from index
INFO[2023-11-06T13:31:30+01:00] cleanup: 2010-10-17/20101017-143552 would be removed from index
INFO[2023-11-06T13:31:30+01:00] cleanup: 2010-10-18/20101018-111318 would be removed from index
INFO[2023-11-06T13:31:30+01:00] cleanup: 2010-10-28/20101028-212921-3 would be removed from index
INFO[2023-11-06T13:31:30+01:00] cleanup: 2015-02-15/20150215-170607980 would be removed from index
INFO[2023-11-06T13:31:30+01:00] cleanup: 2015-10-24/20151024-111643183 would be removed from index
INFO[2023-11-06T13:31:30+01:00] cleanup: 2023-05-22/20230522-162634802 would be removed from index
INFO[2023-11-06T13:31:30+01:00] cleanup: 2023-05-22/20230522-164857955 would be removed from index
INFO[2023-11-06T13:31:30+01:00] cleanup: 2023-07-04/20230704-105417748 would be removed from index
INFO[2023-11-06T13:31:30+01:00] cleanup: 2023-07-04/20230704-122031763 would be removed from index
INFO[2023-11-06T13:31:30+01:00] cleanup: 2023-07-22/20230722-185127996 would be removed from index
INFO[2023-11-06T13:31:30+01:00] cleanup: 2023-08-31/20230831-184834381 would be removed from index

```

**Reply by lastzero** [reply]
Can you check what files these are, if they are broken/readable and if there could be any reason for this eg they are ignored with a ppignore file or have been stacked with other files (which is the most common cause)?

**Reply by lastzero** [reply]
Other than that, you should let cleanup do it's work and see if it helps. Note you can use it from the UI at any time by checking the option in Library > Index. The description says it deletes orphaned index entries, so exactly what seems to be the case here.
